haystack/core/pipeline/breakpoint.py

```diff
@@ -486,93 +444,24 @@ def _create_pipeline_snapshot_from_tool_invoker(
             },
         },
     )
-    if parent_snapshot is None:
-        # Create an empty pipeline snapshot if no parent snapshot is provided
-        final_snapshot = PipelineSnapshot(
-            pipeline_state=PipelineState(inputs={}, component_visits={}, pipeline_outputs={}),
-            timestamp=agent_snapshot.timestamp,
-            break_point=agent_snapshot.break_point,
-            agent_snapshot=agent_snapshot,
-            original_input_data={},
-            ordered_component_names=[],
-            include_outputs_from=set(),
-        )
-    else:
-        final_snapshot = replace(parent_snapshot, agent_snapshot=agent_snapshot)
-
-    return final_snapshot
-
 
-def _trigger_chat_generator_breakpoint(*, pipeline_snapshot: PipelineSnapshot) -> None:
-    """
-    Trigger a breakpoint before ChatGenerator execution in Agent.
-
-    :param pipeline_snapshot: PipelineSnapshot object containing the state of the pipeline and Agent snapshot.
-    :raises BreakpointException: Always raised when this function is called, indicating a breakpoint has been triggered.
-    """
-    if not isinstance(pipeline_snapshot.break_point, AgentBreakpoint):
-        raise ValueError("PipelineSnapshot must contain an AgentBreakpoint to trigger a chat generator breakpoint.")
+    # Create an empty pipeline snapshot
+    return PipelineSnapshot._from_agent_snapshot(agent_snapshot=agent_snapshot)
 
-    if not isinstance(pipeline_snapshot.agent_snapshot, AgentSnapshot):
-        raise ValueError("PipelineSnapshot must contain an AgentSnapshot to trigger a chat generator breakpoint.")
 
-    break_point = pipeline_snapshot.break_point.break_point
-    full_file_path = _save_pipeline_snapshot(pipeline_snapshot=pipeline_snapshot)
-    msg = (
-        f"Breaking at {break_point.component_name} visit count "
-        f"{pipeline_snapshot.agent_snapshot.component_visits[break_point.component_name]}"
-    )
-    raise BreakpointException(
-        message=msg,
-        component=break_point.component_name,
-        pipeline_snapshot=pipeline_snapshot,
-        pipeline_snapshot_file_path=full_file_path,
-    )
-
-
-def _trigger_tool_invoker_breakpoint(*, llm_messages: list[ChatMessage], pipeline_snapshot: PipelineSnapshot) -> None:
+def _should_trigger_tool_invoker_breakpoint(break_point: ToolBreakpoint, llm_messages: list[ChatMessage]) -> bool:
```

Comment:
It seems there are no unit test for this - but maybe implicit testing is fine. WDYT?

Reply:
I think implicit tests are fine, we have quite a few tests that do test this implicitly